core/variant/variant_parser.h

```diff
@@ -60,13 +61,17 @@ class VariantParser {
 	};
 
 	struct StreamFile : public Stream {
+	private:
+		uint64_t buffer_read_end_position;
```

Comment:
> It seems to be depending on the assumption that the file pointer and the `VariantParser` are in sync, and this will not be the case with a readahead. And seeking within the file outside the `VariantParser` will also mess up the sync.

It should be possible to use this to guarantee that the position of the file in the stream is not affected by external seek calls.

Reply:
> If `f` is replaced, something may go wrong.

Yes, this is what I'm concerned about, a `f->seek()` then assuming the `VariantParser` will be in sync. This could require e.g. a call to the `VariantParser` to dirty the cache, but there are a few possible bugs. Which is why I'm suspecting just as a get started we should have a PR to optionally turn off the readahead (which I'm just doing) then we can spend some time polishing up your PR at our leisure.

Reply:
Set it to private, use setter, it should work.

Reply:
In `_read_buffer()`, record the previous position of `f`, and then seek back to the previous position after reading the buffer, `buffer_read_end_position` records the position of f after reading.. The situation is getting more and more complicated. Is it really necessary to do this?